libblkid/src/superblocks/scoutfs.c

```diff
@@ -0,0 +1,123 @@
+/*
+ * Copyright (C) 2025 Versity, Inc.
+ *
+ * This file may be redistributed under the terms of the
+ * GNU Lesser General Public License.
+ */
+
+#include <inttypes.h>
+#include "superblocks.h"
+#include "crc32c.h"
+
+#define SCOUTFS_UUID_BYTES 16
+
+#define SCOUTFS_BLOCK_SM_SHIFT 12
+#define SCOUTFS_BLOCK_SM_SIZE  (1 << SCOUTFS_BLOCK_SM_SHIFT)
+#define SCOUTFS_BLOCK_LG_SHIFT 16
+#define SCOUTFS_BLOCK_LG_SIZE  (1 << SCOUTFS_BLOCK_LG_SHIFT)
+
+struct scoutfs_block_header {
+	uint32_t crc;
+	uint32_t magic;
+	uint64_t fsid;
+	uint64_t seq;
+	uint64_t blkno;
+};
+
+#define SCOUTFS_FLAG_IS_META_BDEV 0x01
+
+struct scoutfs_super_block {
+	struct scoutfs_block_header hdr;
+	uint64_t id;
+	uint64_t fmt_vers;
+	uint64_t flags;
+	uint8_t uuid[SCOUTFS_UUID_BYTES];
+	/* rest omitted */
+};
+
+static int probe_scoutfs(blkid_probe pr, const struct blkid_idmag *mag, int flag)
+{
+	const struct scoutfs_super_block *sb;
+	const unsigned char *buf = NULL;
+	uint32_t crc;
+	sb = blkid_probe_get_sb(pr, mag, struct scoutfs_super_block);
+	if (sb == NULL)
+		return errno ? -errno : 1;
+
+	/* scoutfs_super_block is always in a SCOUTFS_BLOCK_SM_SIZE block */
+	buf = blkid_probe_get_sb_buffer(pr, mag, SCOUTFS_BLOCK_SM_SIZE);
```

Comment:
Missing error handling. Also this now reads from the device twice.
It would be better to pad `struct scoutfs_super_block` to `SCOUTFS_BLOCK_SM_SIZE`.